e2e/e2e_test.go

```diff
@@ -100,6 +102,53 @@ func TestE2ERestartSync(t *testing.T) {
 	delete()
 }
 
+func TestE2ERestartConsul(t *testing.T) {
+	// Test that when Consul restarts, CTS is able to reconnect and react to
+	// changes in Consul catalog.
+	t.Parallel()
+
+	consul := testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
+		HTTPSRelPath: "../testutils",
+	})
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "restart_consul")
+	cleanup := testutils.MakeTempDir(t, tempDir) // cleanup at end if no errors
+
+	configPath := filepath.Join(tempDir, configFile)
+	config := baseConfig().appendConsulBlock(consul).
+		appendTerraformBlock(tempDir).appendDBTask()
+	config.write(t, configPath)
+
+	// start CTS
+	cmd, err := runSync(configPath)
+	require.NoError(t, err)
+	defer stopCommand(cmd)
+	time.Sleep(5 * time.Second)
+
+	// stop Consul
+	consul.Stop()
+	time.Sleep(2 * time.Second)
+
+	// restart Consul
+	consul = testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
+		HTTPSRelPath: "../testutils",
+		PortHTTPS:    parsePort(t, consul.HTTPSAddr),
+	})
```

Comment:
Interesting to find that the SDK doesn't have a `Start()` or restart method for the Consul test server!

I don't like the relative path being a parameter, it's awkward and clunky. But that aside. Would it be possible to get the port from the config?

```suggestion
	consul = testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
		HTTPSRelPath: "../testutils",
		PortHTTPS:   consul.Config.Ports.HTTPS),
	})
```

Reply:
:nod: a `Start()` or `Restart()` would be great. I agree, I don't like the look of the relative path as a config field as much. It seemed like we might add more customizations to the test server though, so I thought the config would be the best option. The other options I thought of seemed even more clunky. If you think of any alternatives you like, I’m very open to it!

Re: getting port from config. Yes!! Thanks for finding this. This is much better. I will update to use this and remove `parsePort`

Reply:
No alternatives come to mind for me just yet.

Reply:
👍 happy to open a new PR if anyone comes up with something better. will continue to mull on this myself as well